### PR TITLE
fix(suite): RBF use increased feeLevel instead of baseFee

### LINKED ISSUE #8863: Fee rate settings ignored when bumping chained transaction
**Describe the bug**
Fee rate settings in send dialogue is  ignored when bumping chained transaction.
Chained transactions are removed but bump includes fee like you are bumping all chained transactions.
See screenshots from BTC and TEST network.
Even i change fee rate to custom Suite ignores it and it counts in all transaction and uses default fee rate.

**Steps to reproduce:**
1. Send transaction to account
2. Send multiple transactions from this account that will include last received input
3. Try to bump first transaction and observe fee rate behavior

**Info:**
 - Suite version: desktop 23.6.1 (bddd516378beadfcff94523e7c94834ba14f147f)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/23.6.1 Chrome/112.0.5615.183 Electron/24.3.1 Safari/537.36
 - OS: MacIntel
 - Screen: 1440x900
 - Device: model T 2.6.1 Universal (revision c3f6e8f19fd647c73b414322e287ba23cf8a0c7c)
 - Transport: BridgeTransport 2.0.33

**Screenshots:**

**BTC**

image

image

image

**TEST**

https://github.com/trezor/trezor-suite/assets/31506317/61d1cc76-13e6-4627-b606-fe8d464942c0

Screenshot 2023-07-05 at 12 35 47
Screenshot 2023-07-05 at 12 34 27
Screenshot 2023-07-05 at 12 30 35

### LINKED ISSUE #9424: error with Bump fee - the suite works with a different value than what I enter
**Describe the bug**
I have submitted the transaction. Subsequently, I wanted to perform a replacement by fee, as I did not set the fee quite ideally, but when the fee is increased after the first step, the Trezor Suite works with a different value than what I enter

**Info:**

-   Suite Version: app
-   OS: Win 10
-   Trezor Suite Version: 23.8.1

**How to reproduce**
First I sent the transaction with 12 sat/B
image

Next click on Bump fee and set the fee to 14 sat/B - then click on replace transaction
image

Subsequently, a window with information about the new transaction will appear - note that the wrong Fee rate value is displayed in the left part - I set 14 sat/B, here it shows 8.62 sat/B. Image already shows the status after signing a new transaction on the wallet - it is possible to replace the transaction
image

After confirming the replacement of the transaction, I get the following warning in the suite and the transaction does not submit. The value displayed for Fee rate in the previous image seems to correspond to the value Trezor Suite was working with, which is not the value I entered in the first step
image

I repeated the procedure with other replacement fee values. When I set the fee to 15 sat/B, the information window (picture number 3) showed me Fee rate 9.62 sat/B, 17 sat/B shows me 11.62 sat/B.

It helped me to enter a value of 19 sat/B, where the suite continued to work with a value of 13.62 sat/B. The transaction was successfully signed and sent. If I look at mempool.space now, the fee shown here is 13.62 sat/B

### PR DESCRIPTION
**Pull request by szymonlesisz**
### First commit
Should fix https://github.com/trezor/trezor-suite/issues/9424

in new fee calculation using `baseFee` parameter returns invalid results when new utxo and new change output are added (described in the issue)

new feeRate for composeTransactions process can be easily taken from the useRbfForm state (increased levels old + new)
it wasn't done like this until now because there was a time when transactions received from the backend didnt have feeRate field, but they have it some time now (there are multiple fallbacks for missing feeRate accross the suite code)

NOTE: field name is ambiguous and confusing (rbfParams.baseFee !== baseFee param in Trezorconnect.composeTransaction) i will remove it completely from the rbfParams in upcoming PR

### Second commit
Should fix https://github.com/trezor/trezor-suite/issues/8863
Display reason and actual calculated fee rate if entered/requested custom fee rate is different than calculated

#### Dropping dust (fee rounding) in send form
Screenshot from 2023-12-05 19-17-15

#### Paying for chained transactions in rbf form

Screenshot from 2023-12-05 20-15-25

### Third commit

do not switch to calculated fee rate if baseFee is used, this would lead to unnecessary overprice tx

### Fourth commit

Reworking chained transactions logic and UI.
Chained txs are divided in to `own` and `others` transactions 

Screenshot from 2023-12-05 19-42-05

Screenshot from 2023-12-05 19-42-52

Screenshot from 2023-12-05 19-43-36

### CODE REVIEW

**Review by marekrjpolak — APPROVED** [review]
Looks good. I haven't checked the updated values in fixtures though.

**Comment by szymonlesisz** [comment]
im waiting for @trezor/qa tests before the merge

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/ChainedTxs.tsx [hunk 1]
```diff
@@ -46,5 +80,16 @@ export const ChainedTxs = ({ txs, network, explorerUrl }: ChainedTxsProps) => (
...
+        {txs.others.length > 0 && (
+            <Label>
+                <Translation id="TR_AFFECTED_TXS_OTHERS" />
+            </Label>
+        )}
+        {txs.others.map(tx => (
+            <StyledTrezorLink key={tx.txid} href={`${explorerUrl}${tx.txid}`} variant="nostyle">
+                <StyledAffectedTransactionItem tx={tx} />
+            </StyledTrezorLink>
+        ))}
```
Comment (komret) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/ChainedTxs.tsx:93]: There is some repeated code here, could be refactored into `ChainedTransactionList` or something.

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/ChangeFee/AffectedTransactions.tsx [hunk 2]
```diff
@@ -57,12 +51,48 @@ const Bullet = styled.div`
     color: ${({ theme }) => theme.TYPE_LIGHT_GREY};
 `;
 
-export const AffectedTransactions = ({ showChained }: { showChained: () => void }) => {
+export const AffectedTransactionItem = ({
```
Comment (komret) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/ChangeFee/AffectedTransactions.tsx]: Could be a separate file. But not sure what is better.

**Review by komret — CHANGES_REQUESTED** [review]

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/ChainedTxs.tsx (hunk 1 above)
Comment (szymonlesisz) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/ChainedTxs.tsx:93]: first loop creates list of `<ChainedTransactionItem>` second creates list of `<StyledAffectedTransactionItem>` 

so the only repeated code is the loop itself :)

therefor potential `ChainedTransactionList` should have additional condition which component to use, i dont see any benefit from it tbh

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/ChangeFee/AffectedTransactions.tsx (hunk 2 above)
Comment (szymonlesisz) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/ChangeFee/AffectedTransactions.tsx]: [7de8809](https://github.com/trezor/trezor-suite/pull/9964/commits/7de88096cdcf762238a57b01177518f0994b77f0)

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/ChainedTxs.tsx (hunk 1 above)
Comment (komret) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/ChainedTxs.tsx:93]: It could take `children`.

**Review by komret — APPROVED** [review]

**Pull request merged**